### ISSUE TITLE
HEVC VR Video not working on laptops

### ISSUE THREAD

**Issue by sourya-sen** [RenderHeads/UnityPlugin-AVProVideo#253]
Problem description:

As title. The file/project/build works fine on Desktops, but doesn't work on laptops. This is a i7-9750m + RTX 2060 laptop so should be sufficient?

Error reads: [AVProVideo] Error: Loading failed.  File not found, codec not supported, video resolution too high or insufficient system resources.

File does play back on Media Player, however.

Device (which devices are you having the issue with - model, OS version number):

Any laptop - tried on a recent Alienware and a Lenovo Legion, both with RTX 2060s.

Media (tell us about your videos - number of videos, resolution, codec, frame-rate, example URLs):

3840x3840 MKV encoded with FB360 encoder from a H265 file. 

System Information:

AVPro Video: plugin v1.10.1 scripts v1.10.1
Target Platform: Standalone
Unity: v2018.4.15f1 WindowsEditor
OS: Desktop - 81SX (LENOVO) - Windows 10  (10.0.0) 64bit - English
CPU: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 12 threads - 16303KB
GPU: NVIDIA GeForce RTX 2060 - NVIDIA - Direct3D 11.0 [level 11.1] - 5980KB - 16384

**Comment by sourya-sen** [comment]
Just to add, HEVC video extensions _is_ installed and the file plays back on Media Player...

**Comment by MorrisRH** [comment]
Would it be possible for you to send us a copy of the video file that isn't working? If so, please send it (or a link to it) to unitysupport@renderheads.com and reference this issue in the email.

**Comment by sourya-sen** [comment]
Hi @MorrisRH as I mentioned in the email I sent a few hours ago, the demo scene of the sphere video with spatial audio that's supplied with the AVPro Video download is not working either on the laptops, so I don't think it's the video file that's the problem. And this behaviour is only seen on all the laptops we've tried (but has worked perfectly fine on all the Desktops we've used)... so it's rather confusing. 

The DirectShow player does play back the file, it's the Media Foundation player that's having this issue on the laptops, but the Direct Show playback is very stuttery so that's not really an option. The HEVC extensions _are_ installed and the video files (both mine and the demo with AVPro) play back fine with Media Player, so not really sure how to troubleshoot further here.

So in short, sending a copy of the video file wouldn't probably help (but I could still try to send it if you'd like - will have to wait till Monday though). Let me know if you need anything else to help solve this as we're very stuck as laptop playback is fairly important for this project.

**Comment by RenderHeadsSte** [comment]
Yes, we will require the video to investigate further.

Being able to playback the video in any other media player is a red herring.  None of us can second guess which API(S) they are using to perform playback.  By way of example, and as you have already seen, the AVPro Video DirectShow implementation does at least playback your video.

Have you tried the video without a FB360 audio track?  Also, have you tried it without the audio track in an MP4 container?  I think it would be good to narrow down exactly what is and isn't the issue here.

**Comment by sourya-sen** [comment]
Haven't tried the video without a FB360 audio track, but MP4 container, both as is as well as with the FB360 container made into an MKV file works.

I'll try the raw HEVC file once just to be sure and I'll send you the video file (at least a small snippet of it, bound by NDAs w/ the project to be able to share the entire thing) **first thing on Monday**. In the meanwhile, as I mentioned, the demo file that is provided with AVPro Video also does not work so that could be a starting point for troubleshooting?

I just find it very odd that it works fine on desktop and not laptops.

**Comment by sourya-sen** [comment]
@RenderHeadsSte quick update: HEVC mp4 file without the audio also does not play back on the laptops. Nor does the Demo_360_StereoSpatialAudio scene/file provided with AVPro (all other scenes/files work fine).

I've sent an email with a small clip to test with. Let me know if you need anything else.

**Comment by MorrisRH** [comment]
Thanks for the clip. It played back with no issue on AMD hardware, however with nVidia hardware (laptop and desktop) it consistently crashed in hevcdecoder_store.dll trying to create a d3d11 surface. Re-encoding with ffmpeg resolved this though.

I'm not sure this is the issue you're having though. If you could try re-encoding the video file using the following ffmpeg command and let is know the results:
`ffmpeg -i <infile> -c:v libx265 -crf 20 -tag:v hvc1 -c:a copy <outfile>`

Could you send on the editor.log file from one of the laptops? It may be able to provide some more clues as to what's going on.

**Comment by sourya-sen** [comment]
Hi @MorrisRH no go with the converted file w/ ffmpeg either. I've sent the editor.log file via email. Thanks again for looking into this.

**Comment by MorrisRH** [comment]
The only thing that explains what you're seeing is if the HEVC codec isn't available.
Was the HEVC codec installed for a different user account? As the codec comes from the Windows Store it is only available to the user whose account was used to install it. Any other users will also have to sign in to the store and install it.

**Comment by sourya-sen** [comment]
Oh wow, that was it! Making a new user account with a separate Microsoft account and logging onto the Store with the same account fixed it!

I was breaking my head over 7 days for this - even made sure the HEVC codes were installed inside the Windows folders, etc... 

Thanks a LOT for helping troubleshoot this, specially when the problem was so completely unrelated to your plugin, much much appreciated. :)
